src/adaptors/curve/index.js

```diff
@@ -1,269 +1,267 @@
+      return coin.usdPrice;
+    }
   }
+};
 
-  dataTvl = dataTvl.filter((el) =>
-    Object.keys(pools.tokenMapping[chainString]).includes(el.pool)
+const main = async () => {
+  // request promises
+  const gaugePromise = getGaugesByChain();
+  const extraRewardPromise = getMainPoolGaugeRewards();
+  const blockchainToPoolSubgraphPromise = Object.fromEntries(
+    BLOCKCHAINIDS.map((blockchainId) => [
+      blockchainId,
+      getSubGraphData(blockchainId),
+    ])
+  );
+  const blockchainToPoolPromise = Object.fromEntries(
+    BLOCKCHAINIDS.filter(
+      (blockchainId) => blockchainId !== 'ethereum'
+    ).map((blockchainId) => [blockchainId, getPools(blockchainId)])
   );
 
-  return dataTvl;
-};
-
-const buildPool = (entry, chainString) => {
-  const token = pools.tokenMapping[chainString][entry.pool];
-  const id = pools.pools[chainString].find(
-    (el) => el.name === entry.pool
-  )?.swap;
-  let apy = entry.apy < 0 ? 0 : entry.apy;
-  apy = chainString === 'ethereum' ? apy : apy * 100;
+  // we need the ethereum data first for the crv prive and await extra query to CG
+  const ethereumPools = await getPools('ethereum');
+  const crvPrice = getCrvPrice(Object.values(ethereumPools));
+
+  // create feeder closure to fill defillamaPooldata asynchroniously
+  const defillamaPooldata = [];
+  const feedLlama = (poolData, blockchainId) => {
+    const [
+      addressToPool,
+      addressToPoolSubgraph,
+      addressToGauge,
+      gaugeAddressToExtraRewards,
+    ] = poolData;
+    for (const [address, pool] of Object.entries(addressToPool)) {
+      const subgraph = addressToPoolSubgraph[address];
+      const gauge = addressToGauge[blockchainId][pool.gaugeAddress];
+      // one gauge can have multiple (different) extra rewards
+      const extraRewards = gaugeAddressToExtraRewards[pool.gaugeAddress];
+
+      const baseAPY = subgraph ? parseFloat(subgraph.latestDailyApy) : 0
+      const crvAPR =
+        gauge && subgraph ? getPoolAPR(pool, subgraph, gauge, crvPrice) : 0;
+      const extraAPR = extraRewards
+        ? extraRewards.map((reward) => reward.apy).reduce((a, b) => a + b)
+        : 0;
+
+      const underlyingTokens = pool.coins.map((coin) => coin.symbol);
+      const rewardTokens = extraRewards ? extraRewards.map(reward => reward.symbol) : []
+      if (crvAPR) {
+        rewardTokens.push("CRV")
+      }
 
-  const newObj = {
-    // ugly, but some swap pools have same address on diff networks
-    pool: `${id}-${chainString}`,
-    chain: utils.formatChain(chainString),
-    project: 'curve',
-    symbol: utils.formatSymbol(token),
-    tvlUsd: entry.tvl,
-    apy: apy,
+      defillamaPooldata.push({
+        pool: address + '-' + blockchainId,
+        chain: utils.formatChain(blockchainId),
+        project: 'curve',
+        underlyingTokens,
+        symbol: underlyingTokens.join('-'),
+        tvlUsd: pool.usdTotal,
+        apy: baseAPY + crvAPR + extraAPR,
+        apyFee: baseAPY,
```

Comment:
Everything looks fine, thank you.
One smol fix apyFee => apyBase
Because the naming changed few days ago (check here https://github.com/DefiLlama/yield-server#adaptors)

Reply:
thx for pointing out the docs. I'm still in the beginning of exploring the project. I've adapted everything accordingly.

Speaking of specification: in the legacy adapter, the chain name is appended to the pool address (as I've done). I'm wondering: in the docs this is not done and in the UI, the column "Chains" seems to indicate an x-to-many relationship. What's up with that? 
![image](https://user-images.githubusercontent.com/98422803/179240139-a980b835-d666-495c-b099-3a5f01388fa9.png)

Reply:
The test fails if `apy` is not included in the pool object. Could it be that either
a) the test is outdated
or 
b) the docs have an error?